src/native-stack/views/NativeStackView.tsx

```diff
@@ -304,6 +307,15 @@ const RouteView = ({
           target: route.key,
         });
       }}
+      onHeaderHeightChange={(e) => {
+        navigation.emit({
+          type: 'headerHeightChange',
+          data: { newHeight: e.nativeEvent.newHeight },
+          target: route.key,
+        });
+
+        setHeaderHeight(e.nativeEvent.newHeight);
```

Comment:
Won't it cause many many rerenders of the whole `Screen` each time the height changes and even if the user did not subscribe to those changes?

Reply:
Hmm, that's a good catch 🤔 I'll reconsider it!